ethereum_history_api/circuits/lib/src/rlp/types_test.nr

```diff
@@ -49,40 +76,160 @@ mod rlp_fragment {
+
+            invalid_length_fragment.assert_eq_u1("Field", rlp, byte)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+        fn invalid_value() {
+            fragment.assert_eq_u1("Field", rlp, 0)
+        }
+    }
+
+    mod assert_eq_u8 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0x11];
+        global fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
+        global value = 17;
 
         #[test]
         fn success() {
-            fragment.assert_eq_bytes32("Field", bytes, bytes)
+            fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP length")]
         fn invalid_length() {
+            let rlp = [0x01, 0x01];
             let invalid_length_fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
 
-            invalid_length_fragment.assert_eq_bytes32("Field", bytes, bytes)
+            invalid_length_fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP value")]
         fn invalid_value() {
-            fragment.assert_eq_bytes32("Field", bytes, alter_array(bytes))
+            fragment.assert_eq_u8("Field", rlp, 16)
+        }
+    }
+
+    mod assert_eq_u32 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0xEE, 0x6B, 0x28, 0x00]; //4000000000
+        global fragment = RlpFragment { offset: 0, length: 4, data_type: STRING };
+        global value = 4000000000;
+
+        #[test]
+        fn success() {
+            fragment.assert_eq_u32("Field", rlp, value)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            fragment.assert_eq_u32("Field", rlp, 0xEE6B28)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+        fn invalid_value() {
+            fragment.assert_eq_u32("Field", rlp, 0x12345678)
+        }
+    }
+
+    mod assert_eq_u64 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0x8A, 0xC7, 0x23, 0x04, 0x89, 0xE8, 0x00, 0x00];
+        global fragment = RlpFragment { offset: 0, length: 8, data_type: STRING };
+        global value = 10000000000000000000;
+
+        #[test]
+        fn success() {
+            fragment.assert_eq_u64("Field", rlp, value)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+         fn invalid_length() {
+            fragment.assert_eq_u64("Field", rlp, 0x8AC723)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+         fn invalid_value() {
+            fragment.assert_eq_u64("Field", rlp, 0x123567812345678)
+        }
+    }
+
+    mod assert_eq_u128 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa];
```

Comment:
Again - why aaaaaaaa? Simplicity. Either use 0x100... or 0xff.... Both of those have obvious meaning as smallest 16 byte value or biggest 16 byte value. 0xaaaaa is some value and begs the question - why was this one choosen